nx/exercises/exercises-1-20.livemd

```diff
@@ -0,0 +1,500 @@
+```
+
+<details style="margin-left: 32px">
+  <summary style="cursor: pointer; font-weight: bold;">Example solution</summary>
+  <div style="padding-top: 8px">
+
+  ```elixir
+  Nx.size(tensor)
+  ```
+
+  </div>
+</details>
+
+<!-- livebook:{"break_markdown":true} -->
+
+**4. Find the number of bytes of memory in `tensor`. (★☆☆)**
+
+```elixir
+tensor = Nx.tensor([[1, 2, 3], [4, 5, 6]])
+# Add your solution here.
+```
+
+<details style="margin-left: 32px">
+  <summary style="cursor: pointer; font-weight: bold;">Example solution</summary>
+  <div style="padding-top: 8px">
+
+  ```elixir
+  Nx.byte_size(tensor)
+  ```
+
+  </div>
+</details>
+
+<!-- livebook:{"break_markdown":true} -->
+
+**5a. Use `Nx.sum/2` to find the sum of all elements of `tensor`. (★☆☆)**
+
+```elixir
+tensor = Nx.tensor([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+# Add your solution here.
+```
+
+<details style="margin-left: 32px">
+  <summary style="cursor: pointer; font-weight: bold;">Example solution</summary>
+  <div style="padding-top: 8px">
+
+  ```elixir
+  Nx.sum(tensor)
+  ```
+
+  </div>
+</details>
+
+<!-- livebook:{"break_markdown":true} -->
+
+**5b. Read the [documentation for `Nx.sum/2`](https://hexdocs.pm/nx/Nx.html#sum/2) then provide the correct option to sum across the _rows_. (★☆☆)**
+
+```elixir
+# Add your solution here.
+```
+
+<details style="margin-left: 32px">
+  <summary style="cursor: pointer; font-weight: bold;">Example solution</summary>
+  <div style="padding-top: 8px">
+
+  ```elixir
+  Nx.sum(tensor, axes: [1])
+  ```
+
+  <div style="padding-top: 16px">
+    <div style="background-color: rgb(62 100 255); color: white; font-weight: bold; border-top-left-radius: 8px; border-top-right-radius: 8px; padding: 8px 16px;">
+      <span style="padding-right: 4px"><i class="ri-lightbulb-line"></i></span>
+      <span>Tip</span>
+    </div>
+    <div style="border-width: 1px; border-color: rgb(62 100 255); border-bottom-left-radius: 8px; border-bottom-right-radius: 8px; padding: 0px 16px;">
+      <p>In Livebook, you can also hover over a function to display its documentation.</p>
+    </div>
+  </div>
+
+  </div>
+</details>
+
+<!-- livebook:{"break_markdown":true} -->
+
+**6. Create a tensor of zeros of size 10 but where the fifth value is 1. (★☆☆)**
+
+```elixir
+# Add your solution here.
+```
+
+<details style="margin-left: 32px">
+  <summary style="cursor: pointer; font-weight: bold;">Example solution</summary>
+  <div style="padding-top: 8px">
+
+  ```elixir
+  zeros = Nx.broadcast(0, {10})
+  index = Nx.tensor([4])
+  Nx.indexed_put(zeros, index, 1)
+  ```
+
+  </div>
+</details>
+
+<!-- livebook:{"break_markdown":true} -->
+
+**7. Create a tensor with values ranging from 10 to 49. (★☆☆)**
+
+```elixir
+# Add your solution here.
+```
+
+<details style="margin-left: 32px">
+  <summary style="cursor: pointer; font-weight: bold;">Example solution</summary>
+  <div style="padding-top: 8px">
+
+  ```elixir
+  Nx.linspace(10, 49, n: 39, type: :s8)
```

Comment:
Would a solution with iota feel more natural? `Nx.iota({39}) |> Nx.add(10)`?

Reply:
I think in Numpy they would use arange or similar, so that solution feels more natural. Perhaps we replace the order of this question with 9.

Reply:
To be clear, make 8 become number 7, number 9 become number 8, and number 7 becomes new number 9 using iota as the answer.

Reply:
[e667514](https://github.com/elixir-nx/nx/pull/1363/commits/e667514a8e1515dbd73cce6dcbde958dfe171fc2)

After reshuffling, I added the iota solution as solution 1 instead of replacing the linspace one (which is now solution 2). The iota one is definitely the more natural solution. But since it requires that extra step, I figured having both approaches was good.

Reply:
Having both is a great call!